Sources/Basic/PathShims.swift

```diff
@@ -65,19 +109,83 @@ public func mkdir(_ path: AbsolutePath, permissions mode: mode_t = S_IRWXU|S_IRW
     guard (rv == 0 || errno == EEXIST) else { throw SystemError.mkdir(errno, path.asString) }
 }
 
-/// Creates a symbolic link at `path` whose content points to `dest`.  If `relative` is true, the symlink contents will be a relative path, otherwise it will be absolute.
+/// Creates a symbolic link at `path` whose content points to `dest`.  If `relative` is true, the symlink contents will be a relative path (by making `dest` relative to `path`, using as many `..` path components as needed); otherwise the symlink path will be absolute.
 public func symlink(_ path: AbsolutePath, pointingAt dest: AbsolutePath, relative: Bool = true) throws {
-    let destString = relative ? dest.relative(to: path.parentDirectory).asString : dest.asString
-    let rv = libc.symlink(destString, path.asString)
-    guard rv == 0 else { throw SystemError.symlink(errno, path.asString, dest: destString) }
+    let dstr = relative ? dest.relative(to: path.parentDirectory).asString : dest.asString
+    let rv = libc.symlink(dstr, path.asString)
+    guard rv == 0 else { throw SystemError.symlink(errno, path.asString, dest: dstr) }
 }
 
-public func rename(_ path: AbsolutePath, to dest: AbsolutePath) throws {
+/// Moves the file system entity at `path` to the new path `dest`.  If there is already a file system entity at `dest`, it is first removed.  Both `path` and `dest` must reside on the same file system.
+public func move(_ path: AbsolutePath, to dest: AbsolutePath) throws {
     let rv = libc.rename(path.asString, dest.asString)
     guard rv == 0 else { throw SystemError.rename(errno, old: path.asString, new: dest.asString) }
 }
 
-public func unlink(_ path: AbsolutePath) throws {
-    let rv = libc.unlink(path.asString)
-    guard rv == 0 else { throw SystemError.unlink(errno, path.asString) }
+/// Removes the file system entity at `path`.  If that was the last reference to the underlying file system object, and no process has the file open, then all resources associated with the file system object are reclaimed.  If `path` is a directory and `recursive` is not true, an `ENOTEMPTY` error is thrown if the directory isn't empty.  If `recursive` is true, the directory contents are recursively removed before the directory itself is removed.  This function doesn't throw an error if `path` doesn't exist.
+public func remove(_ path: AbsolutePath, recursive: Bool = true) throws {
+    // First try removing it as a directory (but treat non-existence as success).
```

Comment:
A couple of reasons:
a) all the other similar functions are currently written in terms of POSIX, and don't rely on Foundation
b) it's hard to vend consistent errors if we use POSIX for some implementations and Foundation for others (since Foundation doesn't vend the underlying `errno`)
c) Foundation is currently different enough between Linux and macOS that it seemed wiser to base the implementation on a single common API, namely POSIX
d) I thought we were trying to limit our use of Foundation until it's a bit more finished, because of the churn